server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -340,7 +340,7 @@ export class TenantManager {
             key2: encryptedTenantKey2,
         };
 
-        const setKeyInCacheSucceeded = await this.setKeyInCache(tenantId, cacheKeys);
+        const setKeyInCacheSucceeded = isCacheEnabled ? await this.setKeyInCache(tenantId, cacheKeys) : false;
```

Comment:
what about other places we call `this.setKeyInCache()` such as lines 427 and 455?

Also, I noticed that `this.setKeyInCache()` already checks if the cache is undefined with `?` this:
```ts
await this.cache?.set(`tenantKeys:${tenantId}`, JSON.stringify(value));
```

So now I am wondering: what was the breaking part that @zhangxin511 mentioned? I might be missing something here, so apologies in advance

Reply:
This was a bad conclusion on me, I saw a build with 
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/3719246/213203049-be588255-5b14-4bfb-b01f-095efd23783e.png">

So I thought the constructor change did't assign a default value and caused this. But the `?` part actually implicitly make the parameter optional and assigned a default value if missing. We can't do both `?` and default `undefined` together actually.

Reply:
I am going to revert back to `cache?`, since it is how the cache was implemented in other places